src/app/url.rs

```diff
@@ -42,4 +44,55 @@ impl App {
             self.open_in_new_window(&url);
         }
     }
+
+    /// Go up one directory in url
+    pub fn go_parent_directory(&self) {
+        let mut parent = String::new();
+
+        if let Some(ref url) = self.webview.widget().get_uri() {
+            if let Some(filename) = get_filename(url) {
+                if filename.is_empty() {
```

Comment:
Do you think it is possible to simply this code by using [`Url::path_segments_mut()`](https://docs.rs/url/newest/url/struct.Url.html#method.path_segments_mut) and [`PathSegmentsMut::pop()`](https://docs.rs/url/newest/url/struct.PathSegmentsMut.html#method.pop) (that might also requires [`PathSegmentsMut::pop_if_empty()`](https://docs.rs/url/newest/url/struct.PathSegmentsMut.html#method.pop_if_empty))?